packages/nextjs/utils/scaffold-stark/contract.ts

```diff
@@ -388,9 +394,9 @@ export function parseParamWithType(
   isRead: boolean,
 ) {
   if (isRead) {
-    if (paramType.includes(CairoTypes.u256)) {
+    if (isCairoU256(paramType)) {
       return tryParsingParamReturnObject(uint256.uint256ToBN, param);
-    } else if (paramType.includes(CairoTypes.ByteArray)) {
+    } else if (isCairoByteArray(paramType)) {
       if (typeof param === "string") {
         return tryParsingParamReturnObject(
           byteArray.byteArrayFromString,
```

Comment:
we use `byteArrayFromString()` when writing, this is `isRead == true ` part of the code. is there any reason why we are using this here, we already had this function below in line 426. @edulanasca

Reply:
likewise for bool

```javascript
else if (isCairoBool(paramType)) {
      return tryParsingParamReturnObject(String, param);
 }
```

Reply:
we are missing `bytes31` , it should return the hex version of the number. this is only for read.
``` javascript
    } else if (isCairoBytes31(paramType)) {
      const fn = (x: bigint) => {
        return `0x${x.toString(16)}`;
      };
      return tryParsingParamReturnObject(fn, param);
    }
```